pkg/runtime/runtime.go

```diff
@@ -1240,6 +1243,21 @@ func (a *DaprRuntime) processComponents() {
 	}
 }
 
+func (a *DaprRuntime) processComponentsOnce() {
+	for {
+		select {
+		case comp := <-a.pendingComponents:
+			if err := a.processOneComponent(comp); err != nil {
```

Comment:
This is probably wrong, as `a.processOneComponent(comp)` is called twice in the flow now.

Reply:
Yeah, this won't work. I don't think it will be called twice, but it won't actually stop the race. It will never actually take anything other than the default case. I had this wrong conceptually in my head.